kernelci/api/helper.py

```diff
@@ -273,6 +273,11 @@ def create_job_node(self, job_config, input_node,
                         runtime=None, platform=None):
         """Create a new job node based on input and configuration"""
         jobfilter = input_node.get('jobfilter')
+        kernel_revision = input_node['data'].get('kernel_revision')
+        if not kernel_revision:
+            print("No kernel revision found in input node")
+            return None
```

Comment:
Why not enclose the `job_node = ...` assignment in a `try: except:` block? This would feel appropriate.

Reply:
Thats always my thoughts, which way more appropriate, generally "safe" .get or try/except. Is there any common practices or something, or it is matter of choice?

Reply:
My feeling is that `.get()` should be used if we accept `None` as a valid value and continue processing even if the key is missing from the dict.

If a missing key is a serious enough issue for us to stop normal processing and return early and/or do something else, then using `var['key']` in a try/except block should be preferred IMHO.

Reply:
Fair enough, will update

Reply:
Actually my suggestion was more to do things this way:
```python
        try:
            job_node = {
                'kind': job_config.kind,
                'parent': input_node['id'],
                'name': job_config.name,
                'path': input_node['path'] + [job_config.name],
                'group': job_config.name,
                'artifacts': {},
                'jobfilter': jobfilter,
                'data': {
                    'kernel_revision': input_node['data']['kernel_revision'],
                },
            }
        except KeyError as key:
            print(f"Input node missing '{key}' attribute, aborting node creation")
            return None
```

Missing `kernel_revision` is a bug, but we could envision a similar bug affecting the `id`, `path` or even `data` attributes, so we could catch all of those with the above solution.

Reply:
agree, updated